src/tools/kdb/cmerge.cpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include "cmerge.hpp"
+#include "kdbmerge.h"
+#include "keyset.hpp"
+#include <cmdline.hpp>
+#include <iostream>
+#include <keysetio.hpp>
+#include <string>
+
+
+CMergeCommand::CMergeCommand ()
+{
+}
+
+CMergeCommand::~CMergeCommand ()
+{
+}
+
+int CMergeCommand::execute (Cmdline const & cl ELEKTRA_UNUSED)
+{
+	if (cl.arguments.size () < 4)
+	{
+		throw invalid_argument ("Wrong number of arguments! At least 4 arguments needed");
+	}
+	kdb::Key oursRoot = cl.createKey (0);
+	kdb::Key theirsRoot = cl.createKey (1);
+	kdb::Key baseRoot = cl.createKey (2);
+	kdb::Key resultRoot = cl.createKey (3);
+	int strategy = MERGE_STRATEGY_ABORT;
+	if (cl.strategy == "preserve")
+	{
+		/** This is here for compatibility. The old merge has preserve as default as defined in cmdline.cpp.
+		 *  As cmerge uses the existing functionality it is still default, even though it does not exist in cmerge.
+		 *  Default in new merge is abort.
+		 */
+		strategy = MERGE_STRATEGY_ABORT;
+	}
+	else if (cl.strategy == "abort")
+	{
+		strategy = MERGE_STRATEGY_ABORT;
+	}
+	else if (cl.strategy == "our")
+	{
+		strategy = MERGE_STRATEGY_OUR;
+	}
+	else if (cl.strategy == "their")
+	{
+		strategy = MERGE_STRATEGY_THEIR;
+	}
+	else if (cl.strategy == "base")
+	{
+		strategy = MERGE_STRATEGY_BASE;
+	}
+	else
+	{
+		throw invalid_argument ("'" + cl.strategy + "' is not a valid strategy. Valid strategies are: abort, our, their, base");
+	}
+
+	kdb::KeySet ours;
+	kdb::KeySet theirs;
+	kdb::KeySet base;
+
+	{
+		kdb::KDB lkdb;
```

Comment:
It would be more efficient to have a single `kdb.get` with the common root.

Reply:
And cutting so that we only get the relevant parts? What if someone wanted to merge something from system with user? Then the root would be /

Reply:
It would be best if you can handle this internally (in elektraMerge). Ideally, it should even work if the 3 key sets actually point to the same KeySet (and only the parentKey differs).

Make sure that elektraMerge **always** have a graceful well-defined behavior, whatever the user passes.

Reply:
I'll add this in a new PR to leave this one in a state where it passed the tests.

Reply:
Yes, good idea. Can you finish up this PR please? (and set "ready to merge")

Reply:
> finish up this PR

means I merge it myself?

Reply:
No, it only means what you did (rebase, check if changes are as wanted, check build server, add "ready to merge").